### PR TITLE
MWEBRCHA-14 : Add view to export contacts to csv file

### PR DESCRIPTION
**Pull request by boulch**
@laulaz 

Can you merge/review this please ? 

Thanking you  ^ ^

### CODE REVIEW

File: src/imio/directory/core/contents/entity/export.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+from zope.component import getSiteManager
+
+import io
+import json
+import pandas
+
+
+ordered_signifiant_columns = [
+    "title",
+    "title_fr",
```
Comment (laulaz) [inline src/imio/directory/core/contents/entity/export.py:16]: This is not a field 😉 
Same for `description_fr` and `subtitle_fr` I think.

File: src/imio/directory/core/contents/entity/export.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
+# # -*- coding: utf-8 -*-
```
Comment (laulaz) [inline src/imio/directory/core/contents/entity/export.py:1]: # # # # # #  😆

File: src/imio/directory/core/contents/entity/export.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    "is_geolocated",
+    "geolocation.latitude",
+    "geolocation.longitude",
+    "mails",
+    "phones",
+    "urls",
+    "private_mails",
+    "private_phones",
+    "private_urls",
+    "private_note",
```
Comment (laulaz) [inline src/imio/directory/core/contents/entity/export.py:45]: Are we sure the rights to `ModifyPortalContent` will always be enough to see private informations ?

File: src/imio/directory/core/contents/entity/export.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        df.to_csv(csv_buffer, index=False, encoding="utf-8")
+
+        self.request.response.setHeader(
+            "Content-Disposition", 'attachment; filename="export.csv"'
+        )
+        self.request.response.setHeader("Content-Type", "text/csv")
+        self.request.response.write(csv_buffer.getvalue())
+        return self.request.response
+
+    def get_directory_response(self, entity_id):
```
Comment (laulaz) [inline src/imio/directory/core/contents/entity/export.py:124]: Old method name I think ... should be renamed

File: src/imio/directory/core/contents/entity/export.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        context_path = "/".join(self.context.getPhysicalPath())
+        query = {
+            "portal_type": "imio.directory.Contact",
+            "path": {"query": context_path, "depth": 1},
```
Comment (laulaz) [inline src/imio/directory/core/contents/entity/export.py:128]: Why use a `depth` of 1 ?  We could have contacts in contacts right ?

File: src/imio/directory/core/contents/entity/export.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                        "imio.directory.vocabulary.Facilities", items["facilities"]
+                    )
+                if attribute == "topics" and getattr(obj, attribute, None) is not None:
+                    items["topics"] = self.get_vocabulary_label(
+                        "imio.smartweb.vocabulary.Topics", items["topics"]
+                    )
+                if attribute == "type" and getattr(obj, attribute, None) is not None:
+                    items["type"] = self.get_vocabulary_label(
+                        "imio.directory.vocabulary.ContactTypes", items["type"]
+                    )
```
Comment (laulaz) [inline src/imio/directory/core/contents/entity/export.py:189]: You could / shoud have used Plone / products serializers : you would have had a lot less work.
Many conversions are already done / tested / optimized for us ...

File: src/imio/directory/core/tests/test_export.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+from plone.formwidget.geolocation.geolocation import Geolocation
+from zope.component import queryMultiAdapter
+
+import io
+import pandas
+import unittest
+from unittest.mock import MagicMock
+
+
+class TestEntity(unittest.TestCase):
```
Comment (laulaz) [inline src/imio/directory/core/tests/test_export.py:17]: Wrong class name

File: src/imio/directory/core/tests/test_export.py [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+    def setUp(self):
+        """Custom shared utility setup for tests"""
+        self.authorized_types_in_entity = ["imio.directory.Contact"]
+        self.unauthorized_types_in_entity = [
+            "imio.directory.Entity",
+            "Folder",
+            "Page",
+            "Link",
+        ]
```
Comment (laulaz) [inline src/imio/directory/core/tests/test_export.py:28]: Not needed here

File: src/imio/directory/core/contents/entity/export.py (hunk 4 above)
Comment (boulch) [reply src/imio/directory/core/contents/entity/export.py]: you right!

File: src/imio/directory/core/contents/entity/export.py (hunk 5 above)
Comment (boulch) [reply src/imio/directory/core/contents/entity/export.py]: Well seen ! 
Set to -1

File: src/imio/directory/core/contents/entity/export.py (hunk 6 above)
Comment (boulch) [reply src/imio/directory/core/contents/entity/export.py:184]: Kamoulox

File: src/imio/directory/core/tests/test_export.py (hunk 7 above)
Comment (boulch) [reply src/imio/directory/core/tests/test_export.py]: shame on me. I change that !

**Review by laulaz — APPROVED** [review]

**Pull request merged**